### PR TITLE
fix: video editor and uploader layout fixes

### PR DESCRIPTION
**Pull request by ArturGaspar**
## Description

Fix video editor/upload layout.

* Video editor spinner before/after:
  editor spinner before editor spinner after
* Video editor thumbnail before/after:
  editor thumb before editor thumb after
* Upload page before/after:
  upload before upload after
* Upload spinner before/after:
  upload spinner before upload spinner after

### Layout changes

1. (Optional) Create a mock video upload API with a useful response delay
   1. Enable mock video uploads (create [waffle flag](http://localhost:18010/admin/waffle/flag/) `contentstore.mock_video_uploads` enabled for everyone)
   2. Modify [videos_post()](https://github.com/openedx/edx-platform/blob/e1bad755651223575bb2039a654a190863a7b1b4/cms/djangoapps/contentstore/video_storage_handlers.py#L710) as follows:
      ```
      def videos_post(course, request):
          """
          ...
          """

          # Delay for observing spinner.
          import time
          time.sleep(20)

          if use_mock_video_uploads():
              return {'files': [{
                  'file_name': 'video.mp4', 'upload_url': 'http://example.com/put_video', 'edx_video_id': '1234'
              }]}, 200

          ...
      ```
2. Open video editor
   1. Enable the new video editor (create [waffle flags](http://localhost:18010/admin/waffle/flag/) `new_core_editors.use_new_video_editor` and `new_core_editors.use_video_gallery_flow` enabled for everyone)
   2. Create a course unit with a video
   3. Edit the video (should open in new editor)
3. See that loading spinner is vertically centered
4. See that video editor thumbnail icon is grey and smaller
5. Open video uploader
   1. Click "< Replace video" to go to the gallery
   2. If there are videos in the gallery and the uploader does not open automatically, click "+ Upload or embed a new video"
6. See the layout changes
7. Upload a video
8. See that loading spinner is vertically centered

### Video uploader close button

1. (Optional) Create a mock video API with useful responses
   1. Enable mock video uploads (create [waffle flag](http://localhost:18010/admin/waffle/flag/) `contentstore.mock_video_uploads` enabled for everyone)
   2. Modify [videos_index_json()](https://github.com/openedx/edx-platform/blob/565b34e4e0904d1a55e56db403893a207c62e4c3/cms/djangoapps/contentstore/video_storage_handlers.py#L692) as follows:
      ```
      def videos_index_json(course):
          """
          ...
          """
          # index_videos, __ = _get_index_videos(course)
          index_videos = [
              {
                  'edx_video_id': 'aaaaaaaa-aaaa-4aaa-aaaa-aaaaaaaaaaaa',
                  'client_video_id': 'video.mp4',
                  'created': '1970-01-01T00:00:00Z',
                  'duration': 42.5,
                  'status': 'upload',
                  'course_video_image_url': 'https://video/images/1234.jpg'
              },
              {
                  'edx_video_id': 'aaaaaaaa-aaaa-4aaa-aaaa-aaaaaaaaaaaa',
                  'client_video_id': 'video.mp4',
                  'created': '1970-01-01T00:00:00Z',
                  'duration': 42.5,
                  'status': 'upload',
                  'course_video_image_url': 'https://video/images/1234.jpg'
              }
          ]
          return JsonResponse({"videos": index_videos}, status=200)
      ```
2. Enable the new video editor (create [waffle flags](http://localhost:18010/admin/waffle/flag/) `new_core_editors.use_new_video_editor` and `new_core_editors.use_video_gallery_flow` enabled for everyone)
3. Create a course unit with a video
4. Open video gallery
   1. Click "Edit" on the video from step 3 (should open in new editor)
   2. Click "< Replace video" to go to the gallery
5. Make sure the uploader is not automatically loaded
6. Click "+ Upload or embed a new video"
7. Click the close (X) button
8. Check that you are taken back to the gallery 
9. Modify the function from step 1 to return an empty list, wait for the application to properly reload:
   ```
   def videos_index_json(course):
       """
       ...
       """
       # index_videos, __ = _get_index_videos(course)
       index_videos = []
       return JsonResponse({"videos": index_videos}, status=200)
   ```
10. Open video gallery (repeat step 4)
11. Make sure the uploader is automatically loaded after the gallery
12. Click the close (X) button
13. Check that you are taken back to the video editing page

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ArturGaspar!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


This is currently a draft pull request. When it is ready for our review and all tests are green, click "Ready for Review", or remove "WIP" from the title, as appropriate.

**Comment by ArturGaspar** [comment]
The coverage loss is only a move of uncovered line 24 to line 17.

Mocking the VideoUploader component and calling the setLoading argument that VideoUploadEditor passes to it should cover it, but I could not figure out how to mock it.

File: src/editors/Editor.jsx [hunk 1]
```diff
@@ -31,9 +31,20 @@ export const Editor = ({
...
-    <div className="d-flex flex-column">
+    <div
+      className="d-flex flex-column"
+      style={{
+        /* Positioned as a proper Paragon FullscreenModal should have been. */
+        position: 'fixed',
+        top: 0,
+        left: 0,
+        right: 0,
+        height: '100%',
```
Comment (farhaanbukhsh) [inline src/editors/Editor.jsx:42]: Using CSS classes seems to be better here https://paragon-openedx.netlify.app/foundations/css-utilities

**Review by farhaanbukhsh — APPROVED** [review]
👍 

- ✅ I tested this on devstack and it looks good apart from the one comment I have
- ✅  I read through the code
- ❌  I checked for accessibility issues
- ❌  Includes documentation
- ❌  I made sure any change in configuration variables is reflected in the corresponding client's `configuration-secure` repository.

File: src/editors/Editor.jsx (hunk 1 above)
Comment (ArturGaspar) [reply src/editors/Editor.jsx:42]: @farhaanbukhsh There is no CSS class for top: 0, left: 0 or right: 0, and I thought leaving everything under style makes it clearer that this entire part is copied from the styling of the Paragon FullscreenModal.

**Comment by tecoholic** [comment]
@navinkarkera Can you kindly add the "jira:2u" label to this PR?

**Comment by e0d** [comment]
@tecoholic is this blended work?

**Comment by mphilbrick211** [comment]
Hi @openedx/teaching-and-learning - this is ready for review!

**Comment by openedx-webhooks** [comment]
I've created issue [TNL-11223](https://2u-internal.atlassian.net/browse/TNL-11223) in the private 2U Jira.

File: src/editors/containers/VideoUploadEditor/VideoUploader.jsx [hunk 2]
```diff
@@ -17,49 +16,51 @@ const URLUploader = () => {
...
-      <div className="x-small align-self-center justify-content-center mx-2 text-dark font-weight-normal">OR</div>
-      <div className="zindex-9 video-id-prompt p-4">
-        <InputGroup className="video-upload-input-group">
+      <div className="small align-self-center justify-content-center mx-2 text-dark font-weight-normal pt-3">
+        OR
+      </div>
+      <div className="zindex-9 video-id-prompt py-3">
+        <InputGroup>
           <FormControl
+            className="m-0"
```
Comment (KristinAoki) [inline src/editors/containers/VideoUploadEditor/VideoUploader.jsx:32]: When the link for a video is extends the full width of the input box, the submit arrow covers the text.

Screenshot 2023-11-30 at 3 38 01 PM

**Review by KristinAoki — CHANGES_REQUESTED** [review]

**Review by KristinAoki — APPROVED** [review]

**Comment by openedx-webhooks** [comment]
@ArturGaspar 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-lib-content-components&entry.867055334=https://github.com/openedx/frontend-lib-content-components/pull/410&entry.1484655318=https://github.com/ArturGaspar&entry.752974735=2023-10-17+14:54&entry.1917517419=2023-12-04+14:58&entry.2133058324=Yes) so we can improve your experience in the future.

**Comment by jesperhodge** [comment]
Good morning @ArturGaspar . I work for the Teaching and Learning team (TNL) at edX and have two questions for you, with context below. The first regards your intentions behind styling decisions you recently merged. The second regards whether you'd like to author a fix, or whether you'd prefer us to.

Regarding this PR, we ran into a problem yesterday that called for immediate attention, as partners working on exam problems were blocked by the absence of vertical scroll bars.
In looking at the PR and changes to Editor.jsx, specifically, you had an exchange with @farhaanbukhsh   in which you appear to be leaning into the Paragon modal styling, in an attempt to be as standard as possible.
It appears to be here that the problem arises, with styling calling for fixed position and some related changes.
Could you please clarify for me your rationale behind changing CSS properties to position: fixed and height: 100%, and the code comment of "Positioned as a proper Paragon FullscreenModal should have been." What problem were you solving, and how is this the solution? I ask because your answer may affect what's an appropriate fix.

We reverted these changes yesterday, to restore problem editor functionality, and are now looking to clean up.

We'd like to have this problem addressed by the end of tomorrow. What's your preference? Would you like to be the one to fix the bug, or would you prefer us to? We are of course happy to review changes you propose if you choose to author the fix.

**Comment by bszabo** [comment]
@ArturGaspar Bernard Szabo here, also from TNL, following up on Jesper's comment from yesterday. The `frontend-app-course-authoring` reverts we put in place to buy time for fixing FLCC have caused a separate set of serious problems, and we urgently need to undo them. TNL is actively working today in the space of this PR to fix FLCC to enable us to bring back all `frontend-app-course-authoring` functionality ASAP.

**Comment by ArturGaspar** [comment]
@jesperhodge Hello. Apologies for the delay in responding. The reason for that decision was that, the component previously used the pgn__modal-fullscreen class, which caused some styilng for the Paragon FullscreenModal to be applied to it. However it did not have the entire inner structure of that modal, which caused the lading spinner to not be vertically centred.

I changed those CSS properties as they were what would have been applied to an inner part of a FullscreenModal, and they fixed the spinner vertical alignment.

**Comment by ArturGaspar** [comment]
@bszabo I see that there were changes to frontend-app-course-authoring as well as to flcc. Is there anything I can still help with?

**Comment by jesperhodge** [comment]
@ArturGaspar Hi Artur, we were able to create a fix to this that dealt with the most critical problems. I'd need to test whether the loading spinner is positioned correctly now - that was not our big priority due to the incident urgency.

**Pull request merged**